packages/filesystem/src/node/nsfw-watcher/nsfw-filesystem-watcher.ts

```diff
@@ -130,30 +129,39 @@ export class NsfwFileSystemWatcherServer implements FileSystemWatcherServer {
                     console.warn(`Failed to watch "${basePath}":`, error);
                     this.unwatchFileChanges(watcherId);
                 }
+            }).then((watcher: nsfw.NSFW | undefined) => {
+                if (!watcher) {
+                    return;
+                }
+                watcher.start().then(() => {
+                    this.options.info('Started watching:', basePath);
+                    if (toDisposeWatcher.disposed) {
+                        if (watcher) {
+                            this.debug('Stopping watching:', basePath);
+                            watcher.stop().then(() => {
+                                // remove a reference to nsfw otherwise GC cannot collect it
+                                watcher = undefined;
+                                this.options.info('Stopped watching:', basePath);
+                                return;
+                            });
+                        }
+                    }
+                    toDisposeWatcher.push(Disposable.create(async () => {
```

Comment:
it is not the same as before. This code should not be executed if `toDisposeWatcher.disposed` is true

Reply:
Thanks for spotting it!